### PR TITLE
Lays the groundwork for organ datumization, makes slime people only have a slime core.

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do

Lays the groundwork for organ datumization with /datum/organ. The most recently inserted organ will take priority for that datum organ function. If an organ is removed say with lung functionality, it will check other organs for lung functionality and replace itself.

Slime people no longer have a heart and lungs, the function of those have been incorporated into the core itself. This means it handles the functions of the heart, lung, and brain all in one. I'm not including eyes and ears in this datumization, because slimes somehow function without them.

If balance team doesnt like the removal of the organs from the slimes, I can remove that part from the PR

## Why It's Good For The Game
#11990 was originally made because you needed lungs and hearts for newcrit, this PR changes that and returns them to their old single-core form. I like the old form, because it was more unique and gave slime people more flavor.

## Images of changes

image

## Testing

Cybernetic lungs can be changed to any type of applicable lung properly. Inserting lungs/hearts works properly if the receiver has space for it. Slimes can use demon hearts without their body failing. Removing the lungs and hearts worked properly, and the organ functions reverted back to the original slime core function. Defibbing works. Viruses with specific internal organ requirements of hearts/lungs work with slimes. Slime people can speak, despite not having a lung organ, just the datum. Slime people can vocal emote. I took a brain out a human, and put a slime brain in their head, and the ABOMINATION was defibbed and worked perfectly fine. Cursed heart deleting itself seems to work fine with the removal of the organ_datum from the human. Chainsawed some people up, some with weird surgery, some just normal, no runtimes.

## Changelog
:cl:
tweak: Slime people now only have a slime core in their head, no hearts or lungs in their chest.
del: Slime hearts and slime lungs.
/:cl:

### CODE REVIEW

File: code/modules/clothing/under/accessories/accessory.dm [hunk 1]
```diff
@@ -149,8 +149,10 @@
 			user.visible_message("[user] places [src] against [user.p_their()] chest and listens attentively.", "You place [src] against your chest...")
 		else
 			user.visible_message("[user] places \the [src] against [M]'s chest and listens attentively.", "You place \the [src] against [M]'s chest...")
-		var/obj/item/organ/internal/H = M.get_int_organ(/obj/item/organ/internal/heart)
-		var/obj/item/organ/internal/L = M.get_int_organ(/obj/item/organ/internal/lungs)
+		var/datum/organ/heart/heurt = M.get_int_organ_datum(ORGAN_DATUM_HEART)
+		var/obj/item/organ/internal/H = heurt.linked_organ
+		var/datum/organ/lungs/loongs = M.get_int_organ_datum(ORGAN_DATUM_LUNGS)
+		var/obj/item/organ/internal/L = loongs.linked_organ
```
Comment (hal9000PR) [inline code/modules/clothing/under/accessories/accessory.dm:155]: whats with the names of these variables

**Review by hal9000PR — COMMENTED** [review]
literally just skimmed this while loading baldurs gate

File: code/modules/clothing/under/accessories/accessory.dm (hunk 1 above)
Comment (DGamerL) [reply code/modules/clothing/under/accessories/accessory.dm:155]: They funny tho

Loooooongs

**Comment by Fox-McCloud** [comment]
Why?

We literally had this back in the early days of Paradise.

It was awful and there was a reason we moved away from it. Having behaviors defined on physical objects that do actual things that can get moved from person to person is vastly superior than having it on a datum.

If you're wanting to abstract organ behavior better from the object, then components are a better use than...this.

File: code/modules/clothing/under/accessories/accessory.dm (hunk 1 above)
Comment (Contrabang) [reply code/modules/clothing/under/accessories/accessory.dm:155]: > whats with the names of these variables

i ran out of variable names please ignore any variables named snorgan, borgan, norgan, or deadgan.
(I can change them back if needed, but it makes it easier to read shit sometimes, even if theyre a bit silly)

**Comment by Contrabang** [comment]
> Why?
> 
> We literally had this back in the early days of Paradise.
> 
> It was awful and there was a reason we moved away from it. Having behaviors defined on physical objects that do actual things that can get moved from person to person is vastly superior than having it on a datum.
> 
> If you're wanting to abstract organ behavior better from the object, then components are a better use than...this.

They're still a part of those objects, they're just abstracted away so that a single object can handle multiple. I haven't touched any that aren't needed for this. 

Components in my personal opinion wouldnt work as well because of the overhead and awkward way you need to access them. This way uses an associated list that allows for procs to easily get the organ datum, which is also held in reference on the human. I could try using elements, but I feel as though that would make them even more awkward. The organ datums are still being mainly handled by the respective organs, its just that a single organ can now have multiple of these features.

File: code/modules/surgery/organs/brain.dm [hunk 2]
```diff
@@ -134,7 +134,8 @@
 	icon = 'icons/mob/slimes.dmi'
 	icon_state = "green slime extract"
 	mmi_icon_state = "slime_mmi"
-//	parent_organ = "chest" Hello I am from the ministry of rubber forehead aliens how are you
+	parent_organ = "chest" // Hello I am from the ministry of rubber forehead aliens how are you
+	organ_datums = list(/datum/organ/heart, /datum/organ/lungs)
```
Comment (Contrabang) [inline code/modules/surgery/organs/brain.dm:139]: This is why I chose the chest in the first place, there was an old funny comment with it. I will undo it if needed.

**Comment by Fox-McCloud** [comment]
Having some organ behavior defined by datums while some are defined by object inheritance is confusing and inconsistent; it's further inconsistent with the nature of shared behaviors that we've previously utilized.

I don't think this is a good idea.

File: code/modules/mob/living/carbon/human/human_organs.dm [hunk 3]
```diff
@@ -162,17 +162,6 @@ old_ue: Set this to a UE string, and this proc will overwrite the dna of organs
...
-Given the name of an organ, returns the external organ it's contained in
-I use this to standardize shadowling dethrall code
--- Crazylemon
-*/
-/mob/living/carbon/human/proc/named_organ_parent(organ_name)
-	if(!get_int_organ_tag(organ_name))
-		return null
-	var/obj/item/organ/internal/O = get_int_organ_tag(organ_name)
-	return O.parent_organ
-
```
Comment (Contrabang) [inline code/modules/mob/living/carbon/human/human_organs.dm:175]: I removed this because it wasnt used anywhere, and it didnt seem particularly useful.

**Comment by Contrabang** [comment]
> Having some organ behavior defined by datums while some are defined by object inheritance is confusing and inconsistent; it's further inconsistent with the nature of shared behaviors that we've previously utilized.
> 
> I don't think this is a good idea.

I could just copy over the code from all the lungs and heart into slime brains and do special snowflake checks to see if someone is a slime person, but I think thats worse. I've used defines like `ORGAN_DATUM_HEART` as arguments for `get_int_organ_datum`, with the two currently only being `ORGAN_DATUM_HEART` and `ORGAN_DATUM_LUNGS`. I can add more documentation to the organ_datum.dm file if needed, explain when and where to use it. But organ code has never been clean, and personally I think some of the changes I've made here have been an improvement. 

Stuff like cybernetic lungs which have multiple breathing options, can now just switch between organ functions instead of it all being hardcoded. Gland hearts and demon hearts no longer needed weird hardcoded checks to make sure they're still functioning, they can just have a heart organ datum subtype that makes sure its always beating.

**Comment by 1080pCat** [comment]
please oh please let me rip slime cores out of slimes on redgrab <3

**Comment by PidgeyThePirate** [comment]
I'd personally love to see some organ weirdness like a species having an additional redundant heart if this becomes a thing.

**Comment by Contrabang** [comment]
This PR failed to pass vote within the allotted timespan, maybe someday

**Pull request closed without merging**